datafusion/core/src/physical_planner.rs

```diff
@@ -689,7 +693,7 @@ impl DefaultPhysicalPlanner {
                         if physical_field.data_type() != logical_field.data_type() {
                             differences.push(format!("field data type at index {} [{}]: (physical) {} vs (logical) {}", i, physical_field.name(), physical_field.data_type(), logical_field.data_type()));
                         }
-                        if physical_field.is_nullable() != logical_field.is_nullable() {
+                        if physical_field.is_nullable() && !logical_field.is_nullable() {
```

Comment:
like it! I still don't get why we still check nullability in schemas equivalence, 🤔 logical a physical schema can be derived differently and nullable sometimes derived in different way as well.

Nullability checks was a source of dozens problems on schema mismatch especially for UNION

Reply:
Likely only a few cases like Union is exception, most of the case doesn't change nullability

Reply:
> logical a physical schema can be derived differently and nullable sometimes derived in different way as well.


agreed, but the earlier delivered schema acts as a contract (promise) for a later delivered schema
if we told the world that `expr` won't contain null values, we can't change the mind at physical planning time. it violates the constraint (promise / contract)
if we told the world that `expr` _may_ contain null values, we didn't promise that it _will_ contain null values, and we may happen to produce no null values (and even be aware of that)